app/internal/notification.py

```diff
@@ -0,0 +1,174 @@
+from operator import attrgetter
+from typing import List, Union
+
+from fastapi import HTTPException
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_406_NOT_ACCEPTABLE
+
+from app.database.models import (
+    Invitation,
+    Message,
+    InvitationStatusEnum,
+    MessageStatusEnum,
+)
+from app.internal.utils import create_model
+
+
+WRONG_NOTIFICATION_ID = (
+    "The notification id you have entered is wrong\n."
+    "If you did not enter the notification id manually, report this exception."
+)
+
+NOTIFICATION_TYPE = Union[Invitation, Message]
+
+UNREAD_STATUS = [
+    InvitationStatusEnum.UNREAD,
+    MessageStatusEnum.UNREAD,
+]
+
+ARCHIVED = [
+    InvitationStatusEnum.DECLINED,
+    MessageStatusEnum.READ,
+]
+
+
+async def get_message_by_id(
+    message_id: int,
+    session: Session,
+) -> Union[Message, None]:
+    """Returns an invitation by an id.
```

Comment:
Leave blank line after the summary line.
The closing `"""` should be in their own line.

See PEP 257 for further details.